Improve local development workflow (#861)
JuliaPackaging/BinaryBuilder.jl: docs/src/jll.md

* `data_txt_path`: this unexported variable is actually equal to `data_txt`, but
  is kept for consistency with all other product types.

## Overriding `dev`'ed JLL packages

> In the event that a user wishes to override the binaries within a JLL package with
> their own binaries, the user may use the `dev_jll()` method inside of JLL packages

Reviewer: ```suggestion
In the event that a user wishes to override the binaries within a JLL package with
their own binaries, the user may use the `dev_jll()` method provided by that JLL package
```

to check out a mutable copy of the package to their `~/.julia/dev` directory.  An
`override` directory will be created within the package directory, providing a

> convenient location for the user to copy in their own libraries over the typical

Reviewer: ```suggestion
convenient location for the user to copy in their own files over the typical
```

I'd also consider changing "binaries" at the start of this paragraph to "files", as in principle one might also want to tweak, say, a header or config file, right? Wrong?
